spacy/strings.pyx

```diff
@@ -123,127 +32,120 @@ cdef class StringStore:
         RETURNS (uint64): The string's hash value.
         """
-        cdef hash_t str_hash
-        if isinstance(string, str):
-            if string in SYMBOLS_BY_STR:
-                return SYMBOLS_BY_STR[string]
-
-            string = string.encode("utf8")
-            str_hash = hash_utf8(string, len(string))
-            self._intern_utf8(string, len(string), &str_hash)
-        elif isinstance(string, bytes):
-            if string in SYMBOLS_BY_STR:
-                return SYMBOLS_BY_STR[string]
-            str_hash = hash_utf8(string, len(string))
-            self._intern_utf8(string, len(string), &str_hash)
-        else:
+        if not isinstance(string, str):
             raise TypeError(Errors.E017.format(value_type=type(string)))
-        return str_hash
 
-    def __len__(self):
-        """The number of strings in the store.
+        if string in SYMBOLS_BY_STR:
+            return SYMBOLS_BY_STR[string]
+        else:
+            return self._intern_str(string)
 
-        RETURNS (int): The number of strings in the store.
+    def as_int(self, string_or_hash: Union[str, int]) -> str:
+        """If a hash value is passed as the input, return it as-is. If the input
+        is a string, return its corresponding hash.
+
+        string_or_hash (str, int): The string to hash or a hash value.
+        RETURNS (int): The hash of the string or the input hash value.
         """
-        return self.keys.size()
+        if isinstance(string_or_hash, int):
+            return string_or_hash
+        else:
+            return get_string_id(string_or_hash)
 
-    def __contains__(self, string_or_id not None):
-        """Check whether a string or ID is in the store.
+    def as_string(self, string_or_hash: Union[str, int]) -> str:
+        """If a string is passed as the input, return it as-is. If the input
+        is a hash value, return its corresponding string.
 
-        string_or_id (str or int): The string to check.
-        RETURNS (bool): Whether the store contains the string.
+        string_or_hash (str, int): The hash value to lookup or a string.
+        RETURNS (str): The stored string or the input string.
         """
-        cdef hash_t str_hash
-        if isinstance(string_or_id, str):
-            if len(string_or_id) == 0:
-                return True
-            elif string_or_id in SYMBOLS_BY_STR:
-                return True
-            str_hash = hash_string(string_or_id)
-        elif _try_coerce_to_hash(string_or_id, &str_hash):
-            pass
+        if isinstance(string_or_hash, str):
+            return string_or_hash
         else:
-            # TODO: Raise an error instead
-            return self._map.get(string_or_id) is not NULL
+            return self._get_interned_str(string_or_hash)
 
-        if str_hash in SYMBOLS_BY_INT:
-            return True
-        else:
-            return self._map.get(str_hash) is not NULL
+    def items(self) -> Collection[Tuple[str, int]]:
```

Comment:
I don't think there's any reason to be vague about the return type here. If it's a list it should just say `List[Tuple...]`.